app/api/toggle-subscribe/route.ts

```diff
@@ -0,0 +1,65 @@
+import { getAppServerSession } from "@/app/auth";
+
+import { NextResponse } from "next/server";
+
+import { prisma } from "../db";
+
+async function toggleSubscribe(name: string, email: string) {
+  console.log(`${name} ${email} subscribes`);
+  const user = await prisma.emailSender.findUnique({
+    where: { email },
+    select: { subscribed: true }
+  });
+  if (user === null) {
+    await prisma.emailSender.create({
+      data: {
+        name,
+        email,
+        subscribed: true
+      }
+    });
+    console.log(`Created user ${name} (${email})`);
+    return { subscribed: true };
+  }
+  if (user.subscribed) {
+    // Unsubscribe
+    await prisma.emailSender.update({
+      where: { email },
+      data: { subscribed: false }
+    });
+    console.log(`Unsubscribed ${name} (${email})`);
+    return { subscribed: false };
+  } else {
+    await prisma.emailSender.update({
+      where: { email },
+      data: { subscribed: true }
+    });
+    console.log(`Subscribed ${name} (${email})`);
+    return { subscribed: true };
+  }
+}
+
+async function getSubscribed(name: string, email: string) {
+  const user = await prisma.emailSender.findUnique({
+    where: { email },
+    select: { subscribed: true }
+  });
+  if (user === null) {
+    return { subscribed: false };
+  }
+  return { subscribed: user.subscribed };
+}
+
+export type LikeEventResponse = Awaited<ReturnType<typeof toggleSubscribe>>;
```

Comment:
Probably rename this to `ToggleSubscribeResponse`?

Reply:
Renamed